src/x86/avx.rs

```diff
@@ -68,6 +70,71 @@ pub unsafe fn _mm256_or_ps(a: f32x8, b: f32x8) -> f32x8 {
     mem::transmute(a | b)
 }
 
+/// Shuffle double-precision (64-bit) floating-point elements within 128-bit
+/// lanes using the control in `imm8`.
+#[inline(always)]
+#[target_feature = "+avx"]
+//#[cfg_attr(test, assert_instr(vshufpd, imm8 = 0x0))] // FIXME
```

Comment:
Should this be enabled now?

Reply:
The [doc](https://gist.github.com/anonymous/25d752fda8521d29699a826b980218fc#file-intel-intrinsics-3-3-15-xml-L14103) says that `vshufpd` should be generated but [clang](https://github.com/llvm-mirror/clang/blob/master/test/CodeGen/avx-builtins.c#L1113-L1117) does not assert that it is the case. Maybe the expanded code is wrong ?
```rust
           match (imm8 >> 0) & 1 {
                0 =>
                match (imm8 >> 1) & 1 {
                    0 =>
                    match (imm8 >> 2) & 1 {
                        0 =>
                        match (imm8 >> 3) & 1 {
                            0 => simd_shuffle4(a, b, [0, 4, 2, 6]),
                            _ => simd_shuffle4(a, b, [0, 4, 2, 7]),
                        },
                        _ =>
                        match (imm8 >> 3) & 1 {
                            0 => simd_shuffle4(a, b, [0, 4, 3, 6]),
                            _ => simd_shuffle4(a, b, [0, 4, 3, 7]),
                        },
                    },
                    _ =>
                    match (imm8 >> 2) & 1 {
                        0 =>
                        match (imm8 >> 3) & 1 {
                            0 => simd_shuffle4(a, b, [0, 5, 2, 6]),
                            _ => simd_shuffle4(a, b, [0, 5, 2, 7]),
                        },
                        _ =>
                        match (imm8 >> 3) & 1 {
                            0 => simd_shuffle4(a, b, [0, 5, 3, 6]),
                            _ => simd_shuffle4(a, b, [0, 5, 3, 7]),
                        },
                    },
                },
                _ =>
                match (imm8 >> 1) & 1 {
                    0 =>
                    match (imm8 >> 2) & 1 {
                        0 =>
                        match (imm8 >> 3) & 1 {
                            0 => simd_shuffle4(a, b, [1, 4, 2, 6]),
                            _ => simd_shuffle4(a, b, [1, 4, 2, 7]),
                        },
                        _ =>
                        match (imm8 >> 3) & 1 {
                            0 => simd_shuffle4(a, b, [1, 4, 3, 6]),
                            _ => simd_shuffle4(a, b, [1, 4, 3, 7]),
                        },
                    },
                    _ =>
                    match (imm8 >> 2) & 1 {
                        0 =>
                        match (imm8 >> 3) & 1 {
                            0 => simd_shuffle4(a, b, [1, 5, 2, 6]),
                            _ => simd_shuffle4(a, b, [1, 5, 2, 7]),
                        },
                        _ =>
                        match (imm8 >> 3) & 1 {
                            0 => simd_shuffle4(a, b, [1, 5, 3, 6]),
                            _ => simd_shuffle4(a, b, [1, 5, 3, 7]),
                        },
                    },
                },
            }
```